src/Microsoft.IdentityModel.Tokens/Validation/ValidationParameters.cs

```diff
@@ -491,12 +491,12 @@ public string RoleClaimType
         public IList<string> ValidIssuers { get; }
 
         /// <summary>
-        /// Gets the <see cref="IList{String}"/> that contains valid types that will be used to check against the JWT header's 'typ' claim.
+        /// Gets or sets the <see cref="IList{String}"/> that contains valid types that will be used to check against the JWT header's 'typ' claim.
         /// If this property is not set, the 'typ' header claim will not be validated and all types will be accepted.
         /// In the case of a JWE, this property will ONLY apply to the inner token header.
         /// The default is <c>null</c>.
         /// </summary>
-        public IList<string> ValidTypes { get; }
+        public IList<string> ValidTypes { get; set; }
```

Comment:
We should not allow this to be set to null.
Use this pattern: https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/blob/ecc2b0813a57da798d361c2edd5e2ae38cfd71f7/src/Microsoft.IdentityModel.Tokens/SecurityTokenDescriptor.cs#L28

Reply:
Is thread safety a concern when validating Token types? I wondering if this is enough to prevent a null setting

public IList<string> ValidTypes
{
    get => _validTypes;
    set => _validTypes = value ?? [];
}

Reply:
We use Interlocked.CompareExchange so that in multiple threads will have the same object.

Reply:
Updated, thanks!